sourcers/json/json.go

```diff
@@ -0,0 +1,62 @@
+package json
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/ardanlabs/conf"
+)
+
+type JSONSourcer struct {
+	m map[string]string
+}
+
+// NewSource returns a conf.Sourcer and, potentially, an error if a
+// read error occurs or the Reader contains an invalid JSON document.
+func NewSource(r io.Reader) (conf.Sourcer, error) {
+	if r == nil {
+		return &JSONSourcer{m: nil}, nil
+	}
+
+	src, err := ioutil.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
+
+	tmpMap := make(map[string]interface{})
```

Comment:
I want you to think about your variable names, why you needed one to being with and what they represent. This doesn't really represent a tmpMap, it represents the data that was read and needs to be processed.